diff --git a/app/views/issues/show.html.erb b/app/views/issues/show.html.erb
index c7cd5689c7067c76b8b44e1a6a92de3de00a8c0e..3c855f5b1c24bab8c22421f44bfe0f013ca310ec 100644
--- a/app/views/issues/show.html.erb
+++ b/app/views/issues/show.html.erb
@@ -122,6 +122,53 @@ end %>
 
 <div id="history">
 <%= render_tabs issue_history_tabs, issue_history_default_tab %>
+
+<% if @issue.notes_addable? %>
+<div class="issue">
+    <%= labelled_form_for @issue, :html => {:id => 'add-notes-form', :multipart => true} do |f| %>
+      <fieldset id="add_note"><legend><%= l(:field_notes) %></legend>
+      <%= f.text_area :notes, :cols => 60, :rows => 10, :class => 'wiki-edit',
+            :data => {
+                :auto_complete => true
+            },
+            :no_label => true %>
+      <%= wikitoolbar_for 'issue_add_note', preview_issue_path(:project_id => @project, :issue_id => @issue) %>
+
+      <% if @issue.safe_attribute? 'private_notes' %>
+      <%= f.check_box :private_notes, :no_label => true %> <label for="issue_private_notes"><%= l(:field_private_notes) %></label>
+      <% end %>
+
+      <%= call_hook(:view_issues_edit_notes_bottom, { :issue => @issue, :notes => @notes, :form => f }) %>
+      </fieldset>
+
+      <fieldset id="add_note_attachments"><legend><%= l(:label_attachment_plural) %></legend>
+        <% if @issue.attachments.any? && @issue.safe_attribute?('deleted_attachment_ids') %>
+        <div class="contextual"><%= link_to l(:label_edit_attachments), '#', :onclick => "$('#existing-attachments').toggle(); return false;" %></div>
+        <div id="note_existing-attachments" style="<%= @issue.deleted_attachment_ids.blank? ? 'display:none;' : '' %>">
+          <% @issue.attachments.each do |attachment| %>
+          <span class="existing-attachment">
+            <%= text_field_tag '', attachment.filename, :class => "icon icon-attachment filename", :disabled => true %>
+            <label>
+              <%= check_box_tag 'issue[deleted_attachment_ids][]',
+                                attachment.id,
+                                @issue.deleted_attachment_ids.include?(attachment.id),
+                                :id => nil, :class => "deleted_attachment" %> <%= l(:button_delete) %>
+            </label>
+          </span>
+          <% end %>
+          <hr />
+        </div>
+        <% end %>
+
+        <div id="note_new-attachments" style="display:inline-block;">
+          <%= render :partial => 'attachments/form', :locals => {:container => @issue} %>
+        </div>
+      </fieldset>
+      <input type="submit" value="<%=l(:button_add)%>" />
+    <% end %>
+</div>
+<% end %>
+
 </div>
 
 <%= render partial: 'action_menu_edit' unless User.current.wants_comments_in_reverse_order? %>
